Fix #4072 and Fix Part of #4938: Revised profile chooser UI

adhiamboperes opened the pull request:
<!-- READ ME FIRST: Please fill in the explanation section below and check off every point from the Essential Checklist! -->
## Explanation
<!--
  - Explain what your PR does. If this PR fixes an existing bug, please include
  - "Fixes #bugnum:" in the explanation so that GitHub can auto-close the issue
  - when this PR is merged.
  -->
Fixes https://github.com/oppia/oppia-android/issues/4072.
The selected image uri created by `com.miui.gallery` app in Xiaomi devices has a different format from the android recommended uri, and is therefore not parsed correctly. Changing the name of the stored file, to be generated using a random UUID, fixes the issue. Please see https://github.com/oppia/oppia-android/issues/4072#issuecomment-2271713809 for more information. There have been no tests added for this bugfix because of issues setting up **ShadowContentResolver** to work with ProfileManagementController.

Fixes Part of https://github.com/oppia/oppia-android/issues/4938: Creates the new Profile chooser UI.
The portrait mode layout is a grid layout recylerview, while landscape mode utilizes a custom recyclerview to create a carousel. This design choice is kind of inconsistent, but more intuitive for users in landscape mode. `ProfileChooserFragmentPresenter` handles the data binding and UI interactions for both layouts. 

Care has been taken to ensure the existing profile behaviour e.g. random color selection and 10 profiles limit have been retained.

Additionally, a new `ParentScreen` enum has been created for the  onboarding IntroActivity to control  when to show the step count. Per PRD, the step count should not show for additional learners.

There have been minor bugfixes to ensure all the user pathways work as detailed in the TDD.

## Essential Checklist
<!-- Please tick the relevant boxes by putting an "x" in them. -->
- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only
<!-- Delete these section if this PR does not include UI-related changes. -->
If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes

### Phone
|LTR|RTL|
|--|--|
|![Screenshot_1739971929](https://github.com/user-attachments/assets/9a39eb7f-c731-4693-99f2-c139c4703567)|![Screenshot_1740172190](https://github.com/user-attachments/assets/e928f796-24a1-4da7-a414-9e43f4683501)|
|![Screenshot_1739971954](https://github.com/user-attachments/assets/1954d63e-5b51-49de-855f-c284c347efba)|![Screenshot_1740172286](https://github.com/user-attachments/assets/cf9d960e-e27c-457b-8ba5-f85fceae10a8)|
|![Screenshot_1740176694](https://github.com/user-attachments/assets/258347de-cd01-4d59-a16a-7427b6935313)|![Screenshot_1740176834](https://github.com/user-attachments/assets/60651f93-f134-4b44-a970-9af2ce7650b1)|
|![Screenshot_1739972307](https://github.com/user-attachments/assets/9f5fa8d1-fb43-4c6a-82b7-2c04d85ce91e)|![Screenshot_1740161079](https://github.com/user-attachments/assets/679c02fd-c194-4a8e-b1ed-b1828d30e885)|
|![Screenshot_1739972674](https://github.com/user-attachments/assets/aa9a6f85-b03d-4b09-b9ef-2e5eae7f8758)|![Screenshot_1740161000](https://github.com/user-attachments/assets/97a8965d-d427-4603-9518-11bb707ea570)|
|![Screenshot_1740160683](https://github.com/user-attachments/assets/f9afdab1-78ac-4898-9fc1-0b60a5edbc77)|![Screenshot_1739972780](https://github.com/user-attachments/assets/655291c2-06fd-42e8-b934-59a33802365b)|


### Tablet
|Portrait|Landscape|
|--|--|
|![Screenshot_1740179588](https://github.com/user-attachments/assets/3e70b57f-da71-44af-8463-ae8aa3b75c0f)|![Screenshot_1740179297](https://github.com/user-attachments/assets/c99a1225-4f59-4162-b336-f92c11ab319e)|
|![Screenshot_1733945372](https://github.com/user-attachments/assets/259935a0-471c-46fc-aa9d-390b959050da)|![Screenshot_1733945392](https://github.com/user-attachments/assets/74e2e51f-ea85-4d97-9226-4d77b72d5a09)|

adhiamboperes committed:
Add kdoc

adhiamboperes committed:
Fix static check failures

adhiamboperes committed:
Fix failing event log events

adhiamboperes committed:
Fix failing app deprecation tests

adhiamboperes committed:
Refactor the app startup state

adhiamboperes committed:
Fix event logs

adhiamboperes committed:
Revert changes to app init

adhiamboperes committed:
Enforce v2 onboarding flow

adhiamboperes committed:
Enforce conditional profile exit for sole learner

adhiamboperes committed:
Revert unnecessary test file changes

adhiamboperes committed:
Fix proto field case

adhiamboperes committed:
Fix SplashActivityTests

adhiamboperes committed:
Merge branch 'develop' into onboarding-profile-type-screen

# Conflicts:
#	scripts/assets/test_file_exemptions.textproto

adhiamboperes committed:
Update test_file_exemptions to the new format

adhiamboperes committed:
Fix indent

adhiamboperes committed:
Merge branch 'onboarding-profile-type-screen' into create-profile-screen

# Conflicts:
#	scripts/assets/test_file_exemptions.textproto

adhiamboperes committed:
Update test_file_exemptions

adhiamboperes committed:
Merge branch 'create-profile-screen' into onboarding-learner-intro-screen

# Conflicts:
#	model/src/main/proto/arguments.proto
#	scripts/assets/test_file_exemptions.textproto

adhiamboperes committed:
Address reviewer comment

adhiamboperes committed:
Merge branch 'onboarding-learner-intro-screen' into onboarding-audio-language-screen

# Conflicts:
#	scripts/assets/test_file_exemptions.textproto

adhiamboperes committed:
Address reviewer comments

adhiamboperes committed:
Populate the language dropdown list

adhiamboperes committed:
Merge branch 'onboarding-audio-language-screen' into onboarding-profile-domain-config

# Conflicts:
#	app/src/main/java/org/oppia/android/app/home/HomeActivity.kt
#	app/src/main/java/org/oppia/android/app/home/HomeFragmentPresenter.kt
#	app/src/main/java/org/oppia/android/app/onboarding/AudioLanguageFragmentPresenter.kt
#	app/src/main/java/org/oppia/android/app/testing/NavigationDrawerTestActivity.kt
#	app/src/test/java/org/oppia/android/app/home/HomeActivityLocalTest.kt
#	scripts/assets/test_file_exemptions.textproto

adhiamboperes committed:
Fix test file exemption path

adhiamboperes committed:
Add missing bazel dep

adhiamboperes committed:
Add dropdown view id

adhiamboperes committed:
Add dropdown view id

adhiamboperes committed:
Hook up app language options

adhiamboperes committed:
Hook up app language options

adhiamboperes committed:
Merge branch 'develop' into onboarding-profile-type-screen

# Conflicts:
#	app/src/main/AndroidManifest.xml
#	app/src/main/java/org/oppia/android/app/activity/ActivityComponentImpl.kt
#	app/src/main/java/org/oppia/android/app/fragment/FragmentComponentImpl.kt
#	app/src/main/res/values-night/color_palette.xml
#	app/src/main/res/values/color_defs.xml
#	app/src/main/res/values/color_palette.xml
#	app/src/main/res/values/component_colors.xml
#	model/src/main/proto/screens.proto
#	utility/src/main/java/org/oppia/android/util/logging/EventBundleCreator.kt

adhiamboperes committed:
Replace png images with svgs

adhiamboperes committed:
Add exemptions for new svgs

adhiamboperes committed:
Pull upstream and fix conflicts

adhiamboperes committed:
Change default avatar bg in v2

adhiamboperes committed:
Fix profile prompt background

adhiamboperes committed:
Merge branch 'develop' into create-profile-screen

# Conflicts:
#	app/src/main/AndroidManifest.xml
#	app/src/main/java/org/oppia/android/app/activity/ActivityComponentImpl.kt
#	app/src/main/java/org/oppia/android/app/fragment/FragmentComponentImpl.kt
#	app/src/main/java/org/oppia/android/app/onboarding/OnboardingProfileTypeFragmentPresenter.kt
#	app/src/main/res/values-night/color_palette.xml
#	app/src/main/res/values/color_palette.xml
#	app/src/main/res/values/component_colors.xml
#	app/src/main/res/values/strings.xml
#	app/src/main/res/values/styles.xml
#	app/src/sharedTest/java/org/oppia/android/app/onboarding/OnboardingProfileTypeFragmentTest.kt
#	model/src/main/proto/screens.proto
#	scripts/assets/test_file_exemptions.textproto
#	utility/src/main/java/org/oppia/android/util/logging/EventBundleCreator.kt

adhiamboperes committed:
Fix deprecated api

adhiamboperes committed:
Merge branch 'create-profile-screen' into onboarding-learner-intro-screen

# Conflicts:
#	app/src/main/res/values/color_defs.xml
#	app/src/main/res/values/color_palette.xml
#	model/src/main/proto/screens.proto

adhiamboperes committed:
Fix kdoc

adhiamboperes committed:
Merge branch 'create-profile-screen' into onboarding-learner-intro-screen

adhiamboperes committed:
Merge remote-tracking branch 'upstream/onboarding-learner-intro-screen' into onboarding-audio-language-screen

adhiamboperes committed:
Fix test_file_exemption

adhiamboperes committed:
Merge branch 'develop' into onboarding-audio-language-screen

# Conflicts:
#	app/src/main/java/org/oppia/android/app/onboarding/IntroFragmentPresenter.kt
#	app/src/main/res/values/styles.xml
#	scripts/assets/test_file_exemptions.textproto

adhiamboperes committed:
Replace ConstraintLayout with FrameLayout

adhiamboperes committed:
Create language dropdown list and the default selection

adhiamboperes committed:
Make otter graphic respond to rtl layouts

adhiamboperes committed:
Add mechanisms to check if a default profile exists: wip

adhiamboperes committed:
Pass the created profileId  and update profile details

Pass the OnboardingFragmentPresenter through intents and bundles

Added a new function to update multiple profile details at once.

adhiamboperes committed:
Refactored IntroFragment arguments to proto, and decorated with the profileId

Passing the profileId to AudioLanguageActivity eliminates the need to fetch it, when required for audio language operations.

adhiamboperes committed:
Prime AudioLanguageActivity to receive a profileId args

Also, remove instances of non-null assertion on internal profileId

adhiamboperes committed:
Fix merge conflicts

adhiamboperes committed:
Refactor to the new central hasProtoExtra

adhiamboperes committed:
Remove redundant non-null assertion

adhiamboperes committed:
Fix duplicating default profile creation

adhiamboperes committed:
Bind audiolanguage pre-selection to the dropdown

adhiamboperes committed:
Fix persist selection on config change

adhiamboperes committed:
Setup audio language pre-selection logic

adhiamboperes committed:
Fix lint errors

adhiamboperes committed:
Cleanup language selection and profile creation

adhiamboperes committed:
Fix missing kdocs

adhiamboperes committed:
Update profile error message view

Use an observable field to dynamically update errors.

adhiamboperes committed:
Refactor IntroActivity intent creation

Pass extras creation to the calling method

adhiamboperes committed:
Update intent tests to reflect new intent extras

adhiamboperes committed:
Add tests for language switching

adhiamboperes committed:
Add helper for creating a default empty profile

adhiamboperes committed:
Update updateNewProfileDetails() kdocs

adhiamboperes committed:
Helper for creating the default empty profile in tests

adhiamboperes committed:
Tests for the new profile update api

adhiamboperes committed:
Additional tests for the profile creation flow

adhiamboperes committed:
Add tests for language selection

Tests for locale, config change, selections and default selection.

adhiamboperes committed:
Fix missing kdoc

adhiamboperes committed:
Fix self-review pass issues.

adhiamboperes committed:
Exempt OnboardingFragmentTest from locale checks

adhiamboperes committed:
Fix failing tests

adhiamboperes committed:
Revert conflicting/already implemented UI layer code

adhiamboperes committed:
Temporary comments

adhiamboperes committed:
Merge branch 'onboarding-language-domain-config' into onboarding-profile-domain-config

# Conflicts:
#	app/src/main/java/org/oppia/android/app/onboarding/IntroActivity.kt
#	app/src/main/java/org/oppia/android/app/onboarding/IntroActivityPresenter.kt
#	app/src/main/java/org/oppia/android/app/onboarding/IntroFragmentPresenter.kt
#	app/src/main/java/org/oppia/android/app/onboarding/OnboardingFragmentPresenter.kt
#	app/src/main/java/org/oppia/android/app/options/AudioLanguageActivity.kt
#	app/src/main/java/org/oppia/android/app/options/AudioLanguageActivityPresenter.kt
#	app/src/main/java/org/oppia/android/app/options/AudioLanguageFragment.kt
#	app/src/main/java/org/oppia/android/app/translation/AppLanguageResourceHandler.kt
#	app/src/sharedTest/java/org/oppia/android/app/onboarding/IntroFragmentTest.kt
#	app/src/sharedTest/java/org/oppia/android/app/options/AudioLanguageFragmentTest.kt
#	app/src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt
#	domain/src/main/java/org/oppia/android/domain/profile/ProfileManagementController.kt
#	domain/src/test/java/org/oppia/android/domain/profile/ProfileManagementControllerTest.kt
#	model/src/main/proto/oppia_logger.proto
#	model/src/main/proto/profile.proto
#	scripts/assets/test_file_exemptions.textproto

adhiamboperes committed:
Fix leftover refactor issue

adhiamboperes committed:
Complete the sole learner onboarding flow

Make sure all expected events are logged, plus tests.

adhiamboperes committed:
Update admin profile profileType in v2

adhiamboperes committed:
Merge branch 'onboarding-language-domain-config' into onboarding-profile-domain-config

# Conflicts:
#	domain/src/main/java/org/oppia/android/domain/profile/ProfileManagementController.kt
#	domain/src/test/java/org/oppia/android/domain/profile/ProfileManagementControllerTest.kt

adhiamboperes committed:
Destroy onboarding activities after onboarding completed.

adhiamboperes committed:
Fix failing SplashActivityTests

adhiamboperes committed:
Fix failing SplashActivityTests + finish migration pathways

adhiamboperes committed:
Onboard existing learner profiles

adhiamboperes committed:
Toggle homescreen with multiple classrooms flag

adhiamboperes committed:
General cleanup

adhiamboperes committed:
Merge branch 'develop' into onboarding-language-domain-config

adhiamboperes committed:
Merge branch 'onboarding-language-domain-config' into onboarding-profile-domain-config

# Conflicts:
#	app/src/main/java/org/oppia/android/app/home/HomeActivity.kt
#	app/src/sharedTest/java/org/oppia/android/app/onboarding/IntroFragmentTest.kt

adhiamboperes committed:
General cleanup

adhiamboperes committed:
add profile selection resources

adhiamboperes committed:
Fix failing test

adhiamboperes committed:
Rename ProfileChooserFragmentPresenter.kt

adhiamboperes committed:
Separate v1 and 2 impls

adhiamboperes committed:
Create the new profile selection screen

adhiamboperes committed:
Cleanup UI

adhiamboperes committed:
Add tests

adhiamboperes commented on app/src/main/java/org/oppia/android/app/profile/ProfileItemViewModel.kt:
```diff
@@ -0,0 +1,6 @@
+import org.oppia.android.app.model.Profile
+import org.oppia.android.app.viewmodel.ObservableViewModel
+
+class ProfileItemViewModel(val profile: Profile) : ObservableViewModel()
```
Add KDocs

adhiamboperes commented on scripts/assets/kdoc_validity_exemptions.textproto:
```diff
@@ -140,6 +140,7 @@ exempted_file_path: "app/src/main/java/org/oppia/android/app/player/stopplaying/
 exempted_file_path: "app/src/main/java/org/oppia/android/app/player/stopplaying/StopStatePlayingSessionListener.kt"
 exempted_file_path: "app/src/main/java/org/oppia/android/app/profile/AddProfileActivity.kt"
 exempted_file_path: "app/src/main/java/org/oppia/android/app/profile/AddProfileActivityPresenter.kt"
+exempted_file_path: "app/src/main/java/org/oppia/android/app/profile/AddProfileListener.kt"
```
Remove

adhiamboperes commented on scripts/assets/kdoc_validity_exemptions.textproto:
```diff
@@ -154,7 +155,7 @@ exempted_file_path: "app/src/main/java/org/oppia/android/app/profile/PinPassword
 exempted_file_path: "app/src/main/java/org/oppia/android/app/profile/ProfileChooserActivity.kt"
 exempted_file_path: "app/src/main/java/org/oppia/android/app/profile/ProfileChooserFragmentPresenter.kt"
-exempted_file_path: "app/src/main/java/org/oppia/android/app/profile/ProfileChooserViewModel.kt"
+exempted_file_path: "app/src/main/java/org/oppia/android/app/profile/ProfileChooserFragmentPresenterV1.kt"
```
Add KDocs and remove from here

adhiamboperes commented on app/src/main/res/layout/profile_selection_fragment.xml:
```diff
@@ -0,0 +1,138 @@
+      android:layout_width="match_parent"
+      android:layout_height="match_parent"
+      app:layout_constraintBottom_toTopOf="@id/profile_chooser_setting_icon"
+      android:layout_marginBottom="64dp"
```
Move all resources to respective files. Verify resources are properly named

adhiamboperes reviewed (commented):
Self review pass 1

adhiamboperes committed:
Merge branch 'develop' into onboarding-language-domain-config

adhiamboperes committed:
Add landscape carousel

adhiamboperes committed:
Address part of reviewer comments

adhiamboperes committed:
Fix lint errors

adhiamboperes committed:
Merge branch 'develop' into onboarding-language-domain-config

adhiamboperes committed:
Merge branch 'onboarding-language-domain-config' into onboarding-profile-domain-config

# Conflicts:
#	app/src/main/java/org/oppia/android/app/onboarding/AudioLanguageFragmentPresenter.kt

adhiamboperes committed:
Merge branch 'onboarding-language-domain-config' of github.com:oppia/oppia-android into onboarding-language-domain-config

adhiamboperes committed:
Merge branch 'onboarding-language-domain-config' into onboarding-profile-domain-config

adhiamboperes committed:
Create test suite for TextInputLayoutBindingAdapters

adhiamboperes committed:
Create test suite for TextInputLayoutBindingAdapters

adhiamboperes committed:
Address more reviewer comments.

adhiamboperes committed:
Address more reviewer comments.

adhiamboperes committed:
Merge branch 'develop' of github.com:oppia/oppia-android into onboarding-language-domain-config

adhiamboperes committed:
Fixed static check failures

adhiamboperes committed:
Merge branch 'onboarding-language-domain-config' into onboarding-profile-domain-config

adhiamboperes committed:
Merge branch 'onboarding-profile-domain-config' into new-profile-chooser-ui-views

adhiamboperes committed:
Fix leftover accidental refactor

adhiamboperes committed:
Fix profile picture upload error

adhiamboperes committed:
Fix static check failures

adhiamboperes committed:
Fix static check failures

adhiamboperes committed:
Fix bazel build errors

adhiamboperes committed:
Fix empty strings assertion

adhiamboperes committed:
Addressed test comments

adhiamboperes committed:
Use proto for app language

adhiamboperes committed:
Refactor language binding logic

adhiamboperes committed:
Allow user to progress on continue clicked.

adhiamboperes committed:
Fix failing test

adhiamboperes committed:
Fix failing audio language tests

adhiamboperes committed:
Fix failing profile creation tests

adhiamboperes committed:
Temporarily remove failing adapter tests

adhiamboperes committed:
Merge branch 'onboarding-language-domain-config' into onboarding-profile-domain-config

# Conflicts:
#	app/src/main/java/org/oppia/android/app/onboarding/AudioLanguageFragmentPresenter.kt
#	app/src/main/java/org/oppia/android/app/onboarding/CreateProfileActivityPresenter.kt

adhiamboperes committed:
Merge branch 'onboarding-profile-domain-config' into new-profile-chooser-ui-views

adhiamboperes committed:
Rename variables for clarity

adhiamboperes committed:
Cleanup PR

adhiamboperes committed:
Remove kdoc exemption

adhiamboperes committed:
Merge branch 'develop' into onboarding-language-domain-config

# Conflicts:
#	app/src/main/java/org/oppia/android/app/player/audio/AudioFragmentPresenter.kt

adhiamboperes committed:
Update profile creation exceptions and tests

adhiamboperes committed:
Properly set up binding adapter tests

adhiamboperes committed:
Merge branch 'onboarding-language-domain-config' into onboarding-profile-domain-config

# Conflicts:
#	domain/src/main/java/org/oppia/android/domain/profile/ProfileManagementController.kt
#	scripts/assets/test_file_exemptions.textproto

adhiamboperes committed:
Fix self review issues

adhiamboperes committed:
Remove unused code

adhiamboperes committed:
Fix nit

adhiamboperes committed:
Merge branch 'develop' into onboarding-language-domain-config

adhiamboperes committed:
Merge branch 'onboarding-language-domain-config' into onboarding-profile-domain-config

adhiamboperes committed:
Remove duplicate exemption

adhiamboperes committed:
Merge branch 'onboarding-profile-domain-config' into new-profile-chooser-ui-views

# Conflicts:
#	app/src/main/res/values/color_defs.xml
#	scripts/assets/test_file_exemptions.textproto

adhiamboperes committed:
Remove leftover profile audiolanguage

adhiamboperes committed:
Fix landscape scrolling behaviour

adhiamboperes committed:
Fix admin profile creation error

adhiamboperes committed:
Fix broken test

adhiamboperes committed:
Merge branch 'develop' into onboarding-language-domain-config

adhiamboperes committed:
Merge branch 'onboarding-language-domain-config' into onboarding-profile-domain-config

adhiamboperes committed:
Integrate classrooms and onboarding

adhiamboperes committed:
Fix Admin profile creation and onboarding

adhiamboperes committed:
Add missing argument proto

adhiamboperes committed:
Merge branch 'onboarding-profile-domain-config' into new-profile-chooser-ui-views

adhiamboperes committed:
Update broken tests

adhiamboperes committed:
Fix profile migration bugs

adhiamboperes committed:
Merge branch 'onboarding-profile-domain-config' into new-profile-chooser-ui-views

adhiamboperes committed:
Hide step count for additional learners onboarding

adhiamboperes committed:
Fix profile pic upload issue

Changed the file naming.

adhiamboperes committed:
Fix failing test

adhiamboperes committed:
Fix failing audio language tests

Vishwajith-Shettigar commented on app/src/main/java/org/oppia/android/app/profile/ProfileListView.kt:
```diff
@@ -0,0 +1,89 @@
+
+  /**
+   * Sets the list of profiles that this view shows.
+   * @param newDataList the new list of profiles to present
```
* @param newDataList the new list of profiles to present.
Add a period at the end of line.

adhiamboperes replied:
We typically only add a period to `@param` lines if they are multiline.

Vishwajith-Shettigar replied:
Okay

Vishwajith-Shettigar commented on app/src/main/java/org/oppia/android/app/profile/ProfileChooserFragmentPresenter.kt:
```diff
@@ -304,4 +321,34 @@ class ProfileChooserFragmentPresenter @Inject constructor(
     }
   }
+
+  /** Handles navigation to the [AdministratorControlsActivity]. */
```
/** Handles navigation to either the [AdministratorControlsActivity] or [AdminAuthActivity]. */ 

The modified KDoc provides more clarity.

adhiamboperes replied:
Updated.

Vishwajith-Shettigar replied:
Okay.

Vishwajith-Shettigar commented on app/src/main/java/org/oppia/android/app/onboarding/IntroActivityPresenter.kt:
```diff
@@ -15,7 +16,7 @@ import javax.inject.Inject
 
 /** Argument key for bundling the profile nickname. */
-const val PROFILE_NICKNAME_ARGUMENT_KEY = "IntroFragment.Arguments"
+const val INTRO_FRAGMENT_ARGUMENT_KEY = "IntroFragment.Arguments"
```
Since you’ve touched this code, would it be possible to consider moving the IntroFragment creation and key into the IntroFragment itself?

adhiamboperes replied:
Done

Vishwajith-Shettigar commented:
@adhiamboperes i have suggested some changes, PTAL. Thanks.

adhiamboperes committed:
Add test for arabic inTextInputLayoutBindingAdaptersTest

adhiamboperes committed:
Refactor formatting

adhiamboperes committed:
Address reviewer comment

adhiamboperes committed:
Cleanup language selection impl

adhiamboperes committed:
Merge branch 'develop' of github.com:oppia/oppia-android into onboarding-language-domain-config

adhiamboperes committed:
Address general comments

adhiamboperes committed:
Nit

adhiamboperes committed:
Nit

adhiamboperes committed:
Merge branch 'onboarding-language-domain-config' into onboarding-profile-domain-config

adhiamboperes committed:
Fix failing tests

adhiamboperes committed:
Fix failing tests

adhiamboperes committed:
Merge branch 'onboarding-profile-domain-config' of github.com:oppia/oppia-android into onboarding-profile-domain-config

adhiamboperes committed:
remove unused import

adhiamboperes committed:
Refactor event log management

adhiamboperes committed:
Merge branch 'develop' of github.com:oppia/oppia-android into onboarding-language-domain-config

adhiamboperes committed:
Refactor function

adhiamboperes committed:
Merge branch 'develop' into onboarding-language-domain-config

adhiamboperes committed:
Merge branch 'develop' into onboarding-language-domain-config

adhiamboperes committed:
Merge branch 'develop' into onboarding-language-domain-config

adhiamboperes committed:
Refactor language selection to use supported languages only

adhiamboperes committed:
Merge branch 'onboarding-language-domain-config' of github.com:oppia/oppia-android into onboarding-language-domain-config

adhiamboperes committed:
Merge branch 'onboarding-language-domain-config' into onboarding-profile-domain-config

adhiamboperes committed:
Refactor test to use existing eventlog util function

adhiamboperes committed:
Null check and readability improvements

adhiamboperes committed:
Refactor test setup

adhiamboperes committed:
Merge branch 'develop' of github.com:oppia/oppia-android into onboarding-profile-domain-config

adhiamboperes committed:
Fix failing tests

adhiamboperes committed:
Fix failing profile chooser tests

adhiamboperes committed:
Resolve merge conflicts

adhiamboperes committed:
Refactor backpress callback

adhiamboperes committed:
Merge branch 'develop' of github.com:oppia/oppia-android into onboarding-profile-domain-config

adhiamboperes committed:
Fix synchronisation issues caused by createEmptyComposeRule()

adhiamboperes committed:
Revert formatting changes

adhiamboperes committed:
Refactor callback code

adhiamboperes committed:
Revert formatting only changes

adhiamboperes committed:
Add missing tests

adhiamboperes committed:
Pull latest changes

adhiamboperes committed:
Merge branch 'onboarding-profile-domain-config' into new-profile-chooser-ui-views

# Conflicts:
#	app/src/main/java/org/oppia/android/app/profile/ProfileChooserFragmentPresenter.kt
#	app/src/sharedTest/java/org/oppia/android/app/onboarding/CreateProfileFragmentTest.kt
#	app/src/sharedTest/java/org/oppia/android/app/profile/ProfileChooserFragmentTest.kt
#	model/src/main/proto/arguments.proto

adhiamboperes committed:
Fix merge issues

adhiamboperes committed:
Merge branch 'onboarding-profile-domain-config' into new-profile-chooser-ui-views

adhiamboperes committed:
Fix lint error

adhiamboperes committed:
Fix missing import

BenHenning commented:
Will need to take a look at this next week.

@theMr17 might you have capacity to take a review pass on this PR?

theMr17 commented on gradle.properties:
```diff
@@ -23,4 +23,3 @@ android.enableJetifier=false
 kotlin.code.style=official
 # Needed to enable Android data binding.
 android.databinding.enableV2=true
-
```
Please revert this formatting change.

adhiamboperes replied:
Done

theMr17 commented on app/src/sharedTest/java/org/oppia/android/app/onboarding/IntroFragmentTest.kt:
```diff
@@ -233,6 +233,7 @@ class IntroFragmentTest {
     ActivityScenario<IntroActivity>? {
       val params = IntroActivityParams.newBuilder()
         .setProfileNickname(testProfileNickname)
+        .setParentScreen(IntroActivityParams.ParentScreen.CREATE_PROFILE_SCREEN)
```
`testProfileNickname` is being indirectly tested by `testFragment_explanationText_isDisplayed`. Is there a way to verify the parent screen as well?

adhiamboperes replied:
1. I discovered that the simplest way to test intent parameters is by using the Fragment that initiates the intent. This ensures all necessary parameters are properly set. As a result, I created `CreateProfileFragmentTest.testFragment_landscape_continueButtonClicked_afterErrorShown_launchesLearnerIntroScreen`.
2. 
3. In `IntroFragmentTest`, the assumption is that all parameters are correctly provided during the test setup, as these are hardcoded in `launchOnboardingLearnerIntroActivity()`. Otherwise, we don’t have a public API available to access those parameters directly.

theMr17 commented on app/src/main/java/org/oppia/android/app/profile/ProfileChooserViewModel.kt:
```diff
@@ -16,7 +17,7 @@ import org.oppia.android.util.data.DataProviders.Companion.toLiveData
 import org.oppia.android.util.locale.OppiaLocale
 import javax.inject.Inject
 
-/** The ViewModel for [ProfileChooserFragment]. */
+/** The ViewModel for [ProfileActionChooserFragment]. */
```
Is the `ProfileActionChooserFragment` going to be introduced in another PR? I couldn't find it.

adhiamboperes replied:
No, I believe this was leftover from an earlier refactor. Removed all occurances.

theMr17 commented on app/src/main/java/org/oppia/android/app/profile/ProfileChooserFragmentPresenterV1.kt:
```diff
@@ -0,0 +1,274 @@
+
+  private fun loginToProfile(profile: Profile) {
+    if (profile.pin.isNullOrBlank()) {
+      profileManagementController.loginToProfile(profile.id).toLiveData().observe(
+        fragment,
+        {
+          if (it is AsyncResult.Success) {
+            if (enableMultipleClassrooms.value) {
+              activity.startActivity(
+                ClassroomListActivity.createClassroomListActivity(activity, profile.id)
+              )
+            } else {
+              activity.startActivity(
+                HomeActivity.createHomeActivity(activity, profile.id)
+              )
+            }
+          }
+        }
+      )
```
The lambda argument can be moved out of the `observe` function. Ditto for other observers in the both the old and new presenters.

adhiamboperes replied:
Done

theMr17 commented on domain/src/main/java/org/oppia/android/domain/profile/ProfileManagementController.kt:
```diff
@@ -857,7 +858,7 @@ class ProfileManagementController @Inject constructor(
     profileType: ProfileType,
     avatarImagePath: Uri?,
-    colorRgb: Int,
+    colorRgb: Int?,
```
Under what circumstances could `colorRgb` be null?

adhiamboperes replied:
We usually provide a default in all cases. Updated to non-null.

theMr17 commented on app/src/main/java/org/oppia/android/app/profile/ProfileChooserFragmentPresenter.kt:
```diff
@@ -155,87 +218,40 @@ class ProfileChooserFragmentPresenter @Inject constructor(
           activity,
           chooserViewModel.adminPin,
-          chooserViewModel.adminProfileId.internalId,
+          0,
```
```suggestion
          profileId = 0,
```
Nit.

adhiamboperes replied:
Updated

theMr17 commented on app/src/main/java/org/oppia/android/app/profile/ProfileChooserFragmentPresenter.kt:
```diff
@@ -77,74 +77,137 @@ class ProfileChooserFragmentPresenter @Inject constructor(
-    )
-    binding.apply {
+
+    binding = ProfileSelectionFragmentBinding.inflate(inflater, container, false).apply {
```
```suggestion
    binding = ProfileSelectionFragmentBinding.inflate(inflater, container, /* attachToRoot= */ false).apply {
```
Nit.

adhiamboperes replied:
Added.

theMr17 commented on app/src/main/res/layout-land/profile_item.xml:
Do we need a separate landscape layout for this? It seems the primary difference lies only in the root layout.

adhiamboperes replied:
Yes, the individual views are constrained differently to the parent and to each other, in both landscape and portrait.

theMr17 commented on app/src/main/res/layout/profile_selection_fragment.xml:
```diff
@@ -0,0 +1,130 @@
+      </androidx.constraintlayout.widget.ConstraintLayout>
+    </androidx.core.widget.NestedScrollView>
+
+    <ImageView
+      android:id="@+id/profile_chooser_setting_icon"
```
From the screenshots, it seems we need to add some left margin to the icon on the portrait layout, perhaps the same amount as the bottom margin.

adhiamboperes replied:
Added a start margin.

theMr17 reviewed (changes requested):
Thanks @adhiamboperes, and apologies for the delayed review. I have left some comments, PTAL.

Additionally, I’m unsure why the _old_ presenter is named `v1`. Shouldn’t it be the other way around?

adhiamboperes committed:
Merge branch 'develop' into new-profile-chooser-ui-views

# Conflicts:
#	app/src/main/java/org/oppia/android/app/drawer/ExitProfileDialogFragment.kt

adhiamboperes committed:
Merge remote-tracking branch 'upstream/develop' into new-profile-chooser-ui-views

adhiamboperes committed:
Merge branch 'develop' into new-profile-chooser-ui-views

# Conflicts:
#	app/src/main/java/org/oppia/android/app/classroom/ClassroomListFragmentPresenter.kt
#	app/src/main/java/org/oppia/android/app/home/HomeFragmentPresenter.kt
#	app/src/main/java/org/oppia/android/app/onboarding/IntroFragmentPresenter.kt
#	app/src/main/java/org/oppia/android/app/profile/ProfileChooserFragmentPresenter.kt
#	app/src/main/java/org/oppia/android/app/splash/SplashActivityPresenter.kt
#	app/src/sharedTest/java/org/oppia/android/app/classroom/ClassroomListFragmentTest.kt
#	app/src/sharedTest/java/org/oppia/android/app/profile/ProfileChooserFragmentTest.kt
#	app/src/sharedTest/java/org/oppia/android/app/splash/SplashActivityTest.kt
#	app/src/test/java/org/oppia/android/app/home/HomeActivityLocalTest.kt
#	domain/src/test/java/org/oppia/android/domain/profile/ProfileManagementControllerTest.kt
#	testing/src/main/java/org/oppia/android/testing/profile/ProfileTestHelper.kt

adhiamboperes committed:
Merge branch 'develop' of github.com:oppia/oppia-android into new-profile-chooser-ui-views

adhiamboperes committed:
Addressed reviewer comments

adhiamboperes committed:
Addressed more reviewer comments

adhiamboperes committed:
Addressed more reviewer comments

adhiamboperes commented:
Thanks for your review @theMr17! I have addressed all your comments. Please let me know of any further questions or changes.
 
Re:

> Additionally, I’m unsure why the old presenter is named v1. Shouldn’t it be the other way around?

The new flow is v2, so we chose to rename the legacy flow with a v1 suffix, and when the feature is complete, we will remove all v1 classes and keep the new ones without a need to rename them.

adhiamboperes commented:
Thanks for your review as well @Vishwajith-Shettigar! I have adressed all of your comments. PTAL and resolve.

Vishwajith-Shettigar reviewed (approved):
@adhiamboperes LGTM.

Vishwajith-Shettigar commented:
> Thanks for your review as well @Vishwajith-Shettigar! I have adressed all of your comments. PTAL and resolve.

I don't see the resolve button, so please feel free to resolve it yourself. Thank you.

adhiamboperes committed:
Fix failing test

adhiamboperes committed:
Fix scroll issue

adhiamboperes committed:
Merge branch 'develop' into new-profile-chooser-ui-views

BenHenning commented on app/src/main/java/org/oppia/android/app/onboarding/IntroFragment.kt:
```diff
@@ -46,7 +54,38 @@ class IntroFragment : InjectableFragment() {
+    const val INTRO_FRAGMENT_ARGUMENT_KEY = "IntroFragment.Arguments"
+
+    /**
+     * Creates a new instance of a IntroFragment.
```
```suggestion
     * Creates a new instance of an [IntroFragment].
```

Slight doc rewording suggestion.

adhiamboperes replied:
Updated.

BenHenning commented on app/src/main/java/org/oppia/android/app/onboarding/IntroFragmentPresenter.kt:
```diff
@@ -43,6 +45,10 @@ class IntroFragmentPresenter @Inject constructor(
 
     profileManagementController.markProfileOnboardingStarted(profileId)
 
+    if (parentScreen == IntroActivityParams.ParentScreen.PROFILE_CHOOSER_SCREEN) {
+      binding.onboardingStepsCount?.visibility = View.GONE
+    }
```
Are both possibilities of this conditional covered in intro fragment tests?

adhiamboperes replied:
Added a test for the `PROFILE_CHOOSER_SCREEN` parent.

BenHenning commented on app/src/main/java/org/oppia/android/app/profile/ProfileListView.kt:
```diff
@@ -0,0 +1,89 @@
+   * Sets the list of profiles that this view shows.
+   * @param newDataList the new list of profiles to present
+   */
+
```
Please remove blank newline.

adhiamboperes replied:
Done

BenHenning commented on domain/src/main/java/org/oppia/android/domain/profile/ProfileManagementController.kt:
```diff
@@ -887,7 +888,7 @@ class ProfileManagementController @Inject constructor(
       } else {
         updatedProfile.avatar =
-          ProfileAvatar.newBuilder().setAvatarColorRgb(colorRgb).build()
+          colorRgb.let { color -> ProfileAvatar.newBuilder().setAvatarColorRgb(color).build() }
```
Why this change? Using the scope function seems unnecessary here and may harm the code's readability.

adhiamboperes replied:
In an earlier commit, I had set `colorRgb` to be nullabe, so this was meant for nullsafety. But I have since refactored the code so that `colorRgb` is not nullable, and I have also reverted this change.

BenHenning commented on domain/src/main/java/org/oppia/android/domain/profile/ProfileManagementController.kt:
```diff
@@ -1225,7 +1226,7 @@ class ProfileManagementController @Inject constructor(
     @Suppress("DEPRECATION") // The code is correct for targeted versions of Android.
     val bitmap = MediaStore.Images.Media.getBitmap(context.contentResolver, avatarImagePath)
-    val fileName = avatarImagePath.pathSegments.last()
+    val fileName = UUID.randomUUID().toString()
```
Since this is fixing a bug, is it possible for us to add a test to at least verify this functionality (even if it's a bit contrived)?

BenHenning replied:
Reminder on this comment since I don't think it was addressed yet.

adhiamboperes replied:
I haven't had much success working with shadowcontentresolver or shadowmediaplayer to resolve this, mainly because  `context.contentResolver.openInputStream(uri)!!` tis used within a private function, so I found it difficult to mock via the public API `updateProfileAvatar`

BenHenning replied:
Are you able to document what you've tried and what hasn't worked in a new issue that we could file to track this? While I'm fine with postponing adding a test for this now, it will eventually become a requirement as we work in the long-term toward 100% code coverage. It'd be nice to capture your efforts so that they don't need to be redone in the future.

adhiamboperes replied:
I filed https://github.com/oppia/oppia-android/issues/5716 to track this(could benefit from a few more details in the solutions tried section)

BenHenning replied:
Sounds good--thanks!

BenHenning commented on app/src/main/java/org/oppia/android/app/profile/ProfileChooserFragmentPresenter.kt:
```diff
@@ -77,74 +78,137 @@ class ProfileChooserFragmentPresenter @Inject constructor(
+    return binding.root
+  }
+
+  private fun ProfileSelectionFragmentBinding.setupPortraitMode() {
```
Nit here & below: the action form should be 'setUp' (as the noun is 'setup').

adhiamboperes replied:
Refactored.

BenHenning commented on app/src/main/java/org/oppia/android/app/profile/ProfileChooserFragmentPresenter.kt:
```diff
@@ -77,74 +78,137 @@ class ProfileChooserFragmentPresenter @Inject constructor(
+      val scrollableWidth = binding.profilesListLandscape?.let {
+        it.width - (it.paddingStart + it.paddingEnd)
+      } ?: 0
+      val offset =
+        if (isLeft) scrollDistance - scrollableWidth else scrollableWidth - scrollDistance
```
Does this work correctly in RTL?

adhiamboperes replied:
With some modification to account for layout direction, it works correctly.
https://github.com/user-attachments/assets/5371b810-e068-48e0-8129-95dff2bee516

BenHenning replied:
This looks really nice! Just to check (since I couldn't quite determine from the video): does RTL mode start with being scrolled all the way to the right? I see that the profile list is correctly reversed (with admin starting at the right), so just wanted to check for initial scroll state.

adhiamboperes replied:
The list is reversed in rtl, so it is not scrolled - users should not be able to see a flicker.

BenHenning replied:
Sorry, I don't think that quite answers my question. The video seems to start with the list of profiles scrolled in the middle, so it's unclear which side of the list the user begins on. Is it the right or left when in RTL?

BenHenning replied:
Per the video that you shared, this does work correctly (there just might be a small graphical bit to fix yet--adding that directly elsewhere).

BenHenning commented on app/src/main/java/org/oppia/android/app/profile/ProfileChooserFragmentPresenter.kt:
```diff
@@ -155,87 +219,40 @@ class ProfileChooserFragmentPresenter @Inject constructor(
           activity,
           chooserViewModel.adminPin,
-          chooserViewModel.adminProfileId.internalId,
+          profileId = 0,
```
Can we avoid hardcoding 0 here?

1. It makes it harder to move away from the default integer 0 being a valid profile ID (#5440).
2. It should eventually be possible for non-ID 0 profiles to become the primary administrator (or, at least, the code should be robust against such a scenario).

adhiamboperes replied:
Updated to use the `adminProfileId` variable available in the viewmodel.

BenHenning commented on app/src/main/java/org/oppia/android/app/profile/ProfileChooserViewModel.kt:
```diff
@@ -26,16 +27,73 @@ class ProfileChooserViewModel @Inject constructor(
+    )
+  }
+
+  private fun retrieveProfiles(profilesResult: AsyncResult<List<Profile>>):
+    List<ProfileItemViewModel> {
```
```suggestion
  private fun retrieveProfiles(
    profilesResult: AsyncResult<List<Profile>>
  ): List<ProfileItemViewModel> {
```

Prefer this format, instead. We generally use this because it avoids needing to indent the entire method body one further indentation compared to all other same-level members in the class (which makes it stand out).

adhiamboperes replied:
Updated, thanks.

BenHenning commented on app/src/sharedTest/java/org/oppia/android/app/profile/ProfileChooserFragmentTest.kt:
```diff
@@ -591,38 +650,280 @@ class ProfileChooserFragmentTest {
+  }
+
+  @Test
+  fun testFragment_enableOnboardingV2_addManyProfiles_checkProfilesSortedAndNoAddProfile() {
```
Is the 'no add profile' bit actually being tested?

Also, I suggest moving that to its own test (so that verifying sorting & the add profile button absence are separate tests).

adhiamboperes replied:
Split the test into 2 and added the assertion for the add profile button.

BenHenning commented on app/src/sharedTest/java/org/oppia/android/app/profile/ProfileChooserFragmentTest.kt:
```diff
@@ -591,38 +650,280 @@ class ProfileChooserFragmentTest {
-    return ProfileChooserActivity
-      .createProfileChooserActivity(ApplicationProvider.getApplicationContext())
+  @Test
+  fun testFragment_enableOnboardingV2_checkAddProfileTextIsDisplayed() {
```
Have you verified that all key functionality from the v1 tests are being appropriately covered in v2 versions? I want to make sure since:

1. It can help ensure correctness as unit tests are good at catching super specific mistakes (e.g. from regressions that the previous implementation has had and fixed).
2. It's valuable to cover old tests since they'll likely be unconditionally deleted during the cleanup PR.

adhiamboperes replied:
Yes, I audited the tests. Tests that should pass for both feature flag on/off are not gated behind the platform parameter. Tests that will be removed are gated ` TestPlatformParameterModule.forceEnableOnboardingFlowV2(false)` while tests that will be kept have ` TestPlatformParameterModule.forceEnableOnboardingFlowV2(true)`. This should be documented in the issue tracking cleanup.

BenHenning reviewed (changes requested):
Thanks @adhiamboperes! Took a full pass. Some top-level feedback:
- The 'add more profiles' and floating action button feel a bit crowded on portrait and would benefit from a bit more space (similar to the landscape version).
- How do the arrows and fling behaviors in landscape behave in RTL mode?
- As discussed during today's CLaM meeting, the landscape layout looks a bit off due to the solo learner bubble being so far left. I'm not quite sure that it should be centered, though, as landscape layouts are usually designed to have important elements easily reachable with one's thumbs. Perhaps placing it at the 1/4 or 1/3 portion of the screen from the left might be a bit cleaner? This seems even worse on tablet landscape mode--that's a lot of blank whitespace.
- As discussed during today's CLaM meeting, how does the new layout behave with Talkback?

adhiamboperes committed:
Merge branch 'develop' of github.com:oppia/oppia-android into new-profile-chooser-ui-views

adhiamboperes committed:
Address reviewer comments.

adhiamboperes committed:
Merge branch 'new-profile-chooser-ui-views' of github.com:oppia/oppia-android into new-profile-chooser-ui-views

adhiamboperes committed:
Address more reviewer comments

theMr17 reviewed (approved):
Apologies for the delayed review. Thanks @adhiamboperes, this LGTM!

adhiamboperes committed:
Merge branch 'develop' into new-profile-chooser-ui-views

BenHenning commented:
@adhiamboperes I think you mentioned in our quick sync today that you'd like another review on this PR. Is that correct, and does that mean that this PR is ready for another pass? I'm asking since it's still assigned to you, so I wasn't sure. :)

adhiamboperes committed:
Merge branch 'develop' of github.com:oppia/oppia-android into new-profile-chooser-ui-views

adhiamboperes commented:
@BenHenning, PTAL.

BenHenning commented on app/src/main/java/org/oppia/android/app/profile/ProfileChooserFragmentPresenter.kt:
```diff
@@ -170,8 +169,25 @@ class ProfileChooserFragmentPresenter @Inject constructor(
+      val isRtl = binding.profilesListLandscape?.let { ViewCompat.getLayoutDirection(it) } ==
+        ViewCompat.LAYOUT_DIRECTION_RTL
+
+      val scrollLeftInRtl = isRtl && isLeft
+      val scrollRightInRtl = isRtl && !isLeft
+      val scrollLeftInLtr = !isRtl && isLeft
+      val scrollRightInLtr = !isRtl && !isLeft
```
Are there tests for verifying this RTL/LTR and directional scrolling behaviors?

adhiamboperes replied:
I have added tests that verify both conditions when using the arrows. Swipe behaviour is covered by existing tests using "scrollTo()".

BenHenning reviewed (changes requested):
Thanks @adhiamboperes! Took another pass. I think most of my previous comments have been addressed, but a few haven't. Besides the open conversations, this includes:

> * The 'add more profiles' and floating action button feel a bit crowded on portrait and would benefit from a bit more space (similar to the landscape version).
> * As discussed during today's CLaM meeting, the landscape layout looks a bit off due to the solo learner bubble being so far left. I'm not quite sure that it should be centered, though, as landscape layouts are usually designed to have important elements easily reachable with one's thumbs. Perhaps placing it at the 1/4 or 1/3 portion of the screen from the left might be a bit cleaner? This seems even worse on tablet landscape mode--that's a lot of blank whitespace.
> * As discussed during today's CLaM meeting, how does the new layout behave with Talkback?

PTAL at the open comment threads and the above.

adhiamboperes committed:
Address reviewer comments re: ui changes

adhiamboperes committed:
Merge branch 'develop' of github.com:oppia/oppia-android into new-profile-chooser-ui-views

adhiamboperes commented:
@BenHenning, PTAL.

I have addressed all pending comments, and update the screenshots in the main PR body. I should be able to upload talkback video tomorrow -- currently facing some physical device issues.

BenHenning commented on app/src/sharedTest/java/org/oppia/android/app/profile/ProfileChooserFragmentTest.kt:
```diff
@@ -712,6 +717,191 @@ class ProfileChooserFragmentTest {
+
+  @Test
+  @Config(qualifiers = "land")
+  fun testFragment_enableOnboardingV2_rtl_checkRightArrowScrollBehaviourIscorrect() {
```
Are this & the one above actually verifying RTL? Only testFragment_enableOnboardingV2_rtl_checkListIsAlphabetical seems to be enabling RTL explicity.

adhiamboperes replied:
I refactored this test to use the Arabic locale.

BenHenning commented on app/src/sharedTest/java/org/oppia/android/app/profile/ProfileChooserFragmentTest.kt:
```diff
@@ -712,6 +717,191 @@ class ProfileChooserFragmentTest {
+
+    launch(ProfileChooserActivity::class.java).use {
+      it.onActivity { activity ->
+        activity.window.decorView.layoutDirection = ViewCompat.LAYOUT_DIRECTION_RTL
```
Does this actually cause the activity to relayout into RTL?

An alternative that might work is switching the language to Arabic since that should force RTL as well, I think.

adhiamboperes replied:
I refactored tests to use the Arabic locale, thanks for the suggestion!

BenHenning commented on app/src/sharedTest/java/org/oppia/android/app/profile/ProfileChooserFragmentTest.kt:
```diff
@@ -712,6 +717,191 @@ class ProfileChooserFragmentTest {
+
+  @Test
+  @Config(qualifiers = "land")
+  fun testFragment_enableOnboardingV2_rtl_checkLeftArrowScrollBehaviourIscorrect() {
```
```suggestion
  fun testFragment_enableOnboardingV2_rtl_checkLeftArrowScrollBehaviourIsCorrect() {
```

Ditto above. Though perhaps we can be more specific than 'correct'?

adhiamboperes replied:
I updated name to `testFragment_enableOnboardingV2_rtl_checkLeftArrowScrollsListToTheRight`, ditto for similar tests.

BenHenning replied:
Much clearer naming, thanks!

BenHenning commented on app/src/sharedTest/java/org/oppia/android/app/profile/ProfileChooserFragmentTest.kt:
```diff
@@ -712,6 +717,191 @@ class ProfileChooserFragmentTest {
+          )
+        )
+
+        testCoroutineDispatchers.runCurrent()
```
Why is this needed? Can't the test just end?

Ditto below.

adhiamboperes replied:
It is not required. I have removed it everywhere.

BenHenning commented on app/src/sharedTest/java/org/oppia/android/app/profile/ProfileChooserFragmentTest.kt:
```diff
@@ -712,6 +717,191 @@ class ProfileChooserFragmentTest {
+            recyclerViewId = R.id.profiles_list_landscape,
+            position = 0,
+            targetViewId = R.id.profile_name_text,
+            itemMatcher = allOf(withText("Admin"), isDisplayed())
```
This doesn't technically verify that it's alphabetical, just that admin is first. Do we already verify elsewhere the whole list is populated correctly?

adhiamboperes replied:
I have refactored the test to verify the position of all the created profiles.

BenHenning commented on app/src/main/res/layout/profile_selection_fragment.xml:
```diff
@@ -100,7 +100,8 @@
       android:layout_marginEnd="@dimen/profile_selection_fragment_add_profile_button_margin_end"
-      android:contentDescription="@string/profile_selection_profile_icon_description"
+      android:layout_marginBottom="@dimen/profile_selection_fragment_add_button_margin_bottom"
+        android:contentDescription="@string/profile_selection_profile_icon_description"
```
Nit: please remove extra indentation here.

adhiamboperes replied:
Done

BenHenning commented on app/src/main/java/org/oppia/android/app/profile/ProfileChooserFragmentPresenter.kt:
```diff
@@ -83,6 +84,16 @@ class ProfileChooserFragmentPresenter @Inject constructor(
+   * Used to retrieve the layout direction that should be used to mirror the direction of the
+   * list based on locale.
+   */
+  @Inject lateinit var resourceHandler: AppLanguageResourceHandler
```
Since this class is a presenter, this can be injected directly via the class's constructor.

adhiamboperes replied:
Refactored.

BenHenning reviewed (changes requested):
> @BenHenning, PTAL.
> 
> I have addressed all pending comments, and update the screenshots in the main PR body. I should be able to upload talkback video tomorrow -- currently facing some physical device issues.

Thanks @adhiamboperes! Took a pass on the new code and left some comments. Just to check, did you address all of my 3 points in the main review message above (quoted from 2 reviews ago)? I did a scan on the images changed in the main PR post, but they don't seem different in any noticeable way and it seems like the original concerns may still be present (minus perhaps Talkback but I don't yet see a video for that to review).

adhiamboperes committed:
Merge branch 'develop' into new-profile-chooser-ui-views

adhiamboperes committed:
Fix check_textview_styles errors

adhiamboperes committed:
Address test feedback

adhiamboperes committed:
Address code feedback

adhiamboperes committed:
Merge branch 'develop' of github.com:oppia/oppia-android into new-profile-chooser-ui-views

adhiamboperes committed:
Fix bugs from manual testing

- adjust profile list header margin top.
- fix exit profile for the sole learner in classroom activity.

adhiamboperes committed:
Fix bug back navigation

Admin profile exits back to PIN activity. Expected destination is the profile chooser. Fixed by finishing pin activity after login.

adhiamboperes committed:
Fix bug list span count

When a profile was deleted, the span count remained set to multiple profiles because of the use of the WasProfileEverAdded property. A single profile should always be center aligned.

adhiamboperes committed:
Fix visible item calculation bug

adhiamboperes committed:
Add min height for accessibility

adhiamboperes committed:
Add min height for accessibility

adhiamboperes committed:
Fix choppy scrolling and scroll to start

adhiamboperes committed:
Fix leftover argument from cleanup

adhiamboperes commented:
> Thanks @adhiamboperes! Took a full pass. Some top-level feedback:
> 
> * The 'add more profiles' and floating action button feel a bit crowded on portrait and would benefit from a bit more space (similar to the landscape version).

I reduced the end margin to create more space between the recyclerview and the button. Also increased the spacing between the button and the prompt.
|Original Layout|Modified Layout|
|---|---|
|![port_multiple](https://github.com/user-attachments/assets/c13767b6-9ec8-4b75-a5fb-fa58da2bd858)|![Screenshot_1740176689](https://github.com/user-attachments/assets/6b79141a-9ff3-4312-937f-b1ba21892f2f)|

> * How do the arrows and fling behaviors in landscape behave in RTL mode?

https://github.com/user-attachments/assets/ace3154e-cff8-444d-9fde-ef3c4eecb623

> * As discussed during today's CLaM meeting, the landscape layout looks a bit off due to the solo learner bubble being so far left. I'm not quite sure that it should be centered, though, as landscape layouts are usually designed to have important elements easily reachable with one's thumbs. Perhaps placing it at the 1/4 or 1/3 portion of the screen from the left might be a bit cleaner? This seems even worse on tablet landscape mode--that's a lot of blank whitespace.

The single profile(or first few profiles), are now placed from the 1/3 part of the screen.
|Original Layout|Modified Layout|
|---|---|
|![land_1](https://github.com/user-attachments/assets/f12b4ea3-e46e-44a4-a83e-b2a02a4c3ce6)|![Screenshot_1739972307](https://github.com/user-attachments/assets/4b790e7b-ad74-4475-aa69-1a605d0911db)|
|![tablet_1](https://github.com/user-attachments/assets/220a9dd9-6377-4318-b7d2-3306de8614b7)|![Screenshot_1740179297](https://github.com/user-attachments/assets/5fdb9575-d682-4700-b1a5-25697e3c38f9)|



> * As discussed during today's CLaM meeting, how does the new layout behave with Talkback?

The layout works as expected both with the arrows and with regular scrolling.
https://github.com/user-attachments/assets/86a954ee-1660-44d4-9718-8fb68862cc7a

adhiamboperes committed:
Fix layout margins.

adhiamboperes committed:
Merge branch 'develop' of github.com:oppia/oppia-android into new-profile-chooser-ui-views

adhiamboperes committed:
Merge branch 'develop' into new-profile-chooser-ui-views

adhiamboperes committed:
Merge branch 'develop' of github.com:oppia/oppia-android into new-profile-chooser-ui-views

adhiamboperes committed:
Merge branch 'new-profile-chooser-ui-views' of github.com:oppia/oppia-android into new-profile-chooser-ui-views

adhiamboperes committed:
merge develop and resolve conflicts

adhiamboperes committed:
Merge branch 'develop' of github.com:oppia/oppia-android into new-profile-chooser-ui-views

adhiamboperes committed:
Fix RTL alignment failures

adhiamboperes committed:
Merge branch 'develop' of github.com:oppia/oppia-android into new-profile-chooser-ui-views

adhiamboperes commented:
@BenHenning, PTAL.

adhiamboperes committed:
Fix formatting

BenHenning reviewed (commented):
Thanks @adhiamboperes! The new spacing looks fantastic, and the Talkback flow looks really good, too!

I think there is just one final thing to address, otherwise the PR LGTM. Specifically: per your excellent video on the RTL behavior, https://m2.material.io/design/usability/bidirectionality.html#localization suggests that the moving arrows need to be reversed (they feel incorrect right now, and I think this is the reason why).

adhiamboperes committed:
Merge branch 'develop' of github.com:oppia/oppia-android into new-profile-chooser-ui-views

adhiamboperes committed:
Cleanup layout and add dynamic margins

adhiamboperes commented:
In the latest commit, I have made some changes that include:

1. Creating dimensions specific to the profile list layout
2. Exporting string literals to the strings file.
3. Removing constraints and creating a dynamic margin decorator instead, using a binding adapter.

|||
|---|---|
|![Screenshot_1742593191](https://github.com/user-attachments/assets/c4650fcd-2689-42cb-846c-e470ac0bdf3d)|![Screenshot_1742850540](https://github.com/user-attachments/assets/cb638280-0e28-42ee-9122-2564e5dc83c5)|
|![Screenshot_1742846355](https://github.com/user-attachments/assets/818f939f-ab34-483b-b7f5-b80a0e55dd76)|![Screenshot_1742850439](https://github.com/user-attachments/assets/d0b7b438-e0aa-4cb7-95da-427429b9380e)|
|![Screenshot_1742846530](https://github.com/user-attachments/assets/52615c20-3b33-48e2-bb34-2d2a4ef8422d)|![Screenshot_1742849009](https://github.com/user-attachments/assets/c8ccda6d-c5c8-4fbc-8a5a-ea32ad798026)|

adhiamboperes committed:
Self review cleanup

adhiamboperes commented:
@BenHenning, PTAL.

adhiamboperes commented:
> Thanks @adhiamboperes! The new spacing looks fantastic, and the Talkback flow looks really good, too!
> 
> I think there is just one final thing to address, otherwise the PR LGTM. Specifically: per your excellent video on the RTL behavior, https://m2.material.io/design/usability/bidirectionality.html#localization suggests that the moving arrows need to be reversed (they feel incorrect right now, and I think this is the reason why).



I had set auto mirrored to be true in the vectors. That has changed now.

![Screenshot_1742846668](https://github.com/user-attachments/assets/e58fafff-9005-4ca1-b31d-4806da3c765e)

adhiamboperes committed:
Merge branch 'develop' into new-profile-chooser-ui-views

BenHenning commented on app/src/main/java/org/oppia/android/app/recyclerview/DynamicMarginItemDecorator.kt:
```diff
@@ -0,0 +1,43 @@
+ * A decorator that assigns a dynamic margin to a recyclerview based on the number of items in the
+ * adapter.
+ */
+class DynamicMarginItemDecorator(
```
I think either the decorator itself needs to be specific to profile selection, or the dimensions need to be generalized. The current general adapter + specific dimensions is a bit confusing.

Feel free to resolve this yourself once addressed.

adhiamboperes replied:
I have thought about this -- creating a generic decorator requires a lot more design, e.g. allowing the desired item count to be customizable, and passing in all the variations of margins desired as well as validating that the provided margins are for the desired boundary.

On the other hand, as is, the decorator supports the specific needs of the profile list. I have opted to rename the class and move the file to the profile package.

BenHenning reviewed (approved):
Thanks @adhiamboperes! Just one comment, otherwise this LGTM!

adhiamboperes committed:
Refactor decorator

adhiamboperes committed:
Merge branch 'develop' of github.com:oppia/oppia-android into new-profile-chooser-ui-views

adhiamboperes committed:
Merge branch 'new-profile-chooser-ui-views' of github.com:oppia/oppia-android into new-profile-chooser-ui-views
